packages/@azure/servicebus/data-plane/lib/core/messageReceiver.ts

```diff
@@ -761,24 +761,41 @@ export class MessageReceiver extends LinkEntity {
           _receiver: this._receiver
         };
         log.error(
-          "[%s] Something is busted. State of Receiver '%s' with address '%s' is: %O",
+          "[%s] Something went wrong. State of Receiver '%s' with address '%s' is: %O",
           connectionId,
           this.name,
           this.address,
           state
         );
       }
       if (shouldReopen) {
+        const disconnectedReceiverName = this.name;
         // provide a new name to the link while re-connecting it. This ensures that
         // the service does not send an error stating that the link is still open.
         const options: ReceiverOptions = this._createReceiverOptions(true);
+
+        this.name = options.name as any;
```

Comment:
I am guessing the reason you have cast as any is because there was a type mismatch?
`this.name` is `string` where as `options.name` is `string|undefined`.
Then the right solution would be
```
if (options.name) {
   this.name = options.name;
}
```

or 

`this.name = options.name!`